### PR TITLE
Chia Blockchain CAIP-2 specification

### PR DESCRIPTION
**Pull request by paninaro**
This PR includes a CAIP-2 specification for identifying blockchains in the Chia ecosystem

### CODE REVIEW

File: chia/caip2.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+```
+# Request
+$ curl -X POST
+       --insecure
+       --cert ~/.chia/mainnet/config/ssl/full_node/private_full_node.crt
+       --key ~/.chia/mainnet/config/ssl/full_node/private_full_node.key
+       -H "Accept: application/json"
+       -H "Content-Type: application/json"
+       https://localhost:8555/get_block_record_by_height
```
Comment (bumblefudge) [inline chia/caip2.md:67]: Is there a testnet fullnode or other public RPC endpoint that can be used to test this without running a fullnode locally? If not, could we link to instructions on setting one up? We generally prefer the CLI examples to be runnable by a casual observer or someone still evaluating whether or not to add Chia to a multi-chain project, so making it dead simple to test/grok/play allows this specification to double as an open invitation to curious researchers to start playing with the RPC themselves...

Comment (bumblefudge) [reply chia/caip2.md:67]: Possible alternative: are there public block explorers with RPC endpoints?

Comment (paninaro) [reply chia/caip2.md:98]: Thanks for the feedback @bumblefudge. Unfortunately it's not possible to issue an RPC request to a random public full node without writing a fair amount of code. I did add a couple of CLI examples for retrieving the genesis challenge using a public Chia blockchain explorer (Spacescan). I also included a link to installation/setup instructions for running a local full node.

**Review by bumblefudge — APPROVED** [review]

**Comment by bumblefudge** [comment]
Great work, thanks!

File: chia/caip2.md (hunk 1 above)
Comment (Yakuhito) [reply chia/caip2.md:98]: @paninaro, I just saw your statement and am truly heartbroken. FireAcademy.io has been running full nodes for well over a year. You can sign up without a credit card, and our free tier includes enough credits that some apps never need to upgrade. Once you generate an API key (for free), calling full node RPC endpoints is as easy as:

```sh
curl -X POST https://kraken.fireacademy.io/[api-key]/leaflet/get_block_record_by_height -H "Content-Type: application/json" -d '{"height": 1337}'
```
Want testnet10 RPC calls? Just switch 'leaflet' to 'leaflet-testnet10':

```sh
curl -X POST https://kraken.fireacademy.io/[api-key]/leaflet-testnet10/get_block_record_by_height -H "Content-Type: application/json" -d '{"height": 1337}'
```

Comment (paninaro) [reply chia/caip2.md:98]: I don't have any philosophical objection to updating the spec to include a fireacademy.io example (sorry for slight @Yakuhito! :)), but as a matter of process, this PR has been sitting around for a few weeks with 1/2 needed approvals. I don't know what the process is with the ChainAgnostic folks, but I'm feeling twitchy about making a change that could reset the clock by another several weeks...

@bumblefudge what would you suggest?

Comment (denisu) [reply chia/caip2.md:98]: There is also Mojonode https://api.mojonode.com/docs

Comment (paninaro) [reply chia/caip2.md:98]: @denisu Mojonode doesn't support testnet10 (currently). I had pinged trepca about this.

Comment (bumblefudge) [reply chia/caip2.md:98]: don't worry about the timing, i'm fine with it as long as the most current, useful, and complete information is there-- remember, your target audience is people who want to hit the ground running quickly and check their work, without necessarily doing a whole developer onboarding!  ideally, someone equipped with only this document could "double check" their work with CLI tools, so feel free to provide multiple ways of doing that.

Comment (paninaro) [reply chia/caip2.md]: Thanks @bumblefudge. I've updated the PR to include examples using FireAcademy.

**Pull request merged**